airflow/docker.go

```diff
@@ -200,18 +199,6 @@ func DockerComposeInit(airflowHome, envFile, dockerfile, imageName string) (*Doc
 //
 //nolint:gocognit
 func (d *DockerCompose) Start(imageName, settingsFile, composeFile, buildSecretString string, noCache, noBrowser bool, waitTime time.Duration, envConns map[string]astrocore.EnvironmentObjectConnection) error {
-	// check if docker is up for macOS
-	containerRuntime, err := GetContainerRuntimeBinary()
-	if err != nil {
-		return err
-	}
-	if runtime.GOOS == "darwin" && containerRuntime == dockerCmd {
-		err := startDocker()
-		if err != nil {
-			return err
-		}
-	}
-
```

Comment:
I moved this to our new pre-run hook, where we do some similar stuff for podman now. That way its in the same place, and we actually cover more scenarios than just `astro dev start` as it is now.